docs/source/tutorial-intro-chipflow-platform.rst

```diff
@@ -0,0 +1,244 @@
+   .. note::
+     If you choose to install ``poetry`` within a venv, ``poetry`` will reuse 
+     that venv instead of creating a new one, so you should ensure that its 
+     version of Python is compatible with the requirements of this project 
+     in ``./pyproject.toml``.
+
+* `Docker <https://docs.docker.com/get-docker/>`_  (or `podman <https://podman.io/getting-started/installation>`_) should be available, it's used for the `dockcross <https://github.com/dockcross/dockcross>`_ builds.
+
+* `openFPGAloader is required <https://trabucayre.github.io/openFPGALoader/guide/install.html>`_ to use a board.
+   * macOS: Easiest way is :bash:`brew install openfpgaloader`.
+   * Linux/Windows: Easiest way may be via the `OSS CAD Suite <https://github.com/YosysHQ/oss-cad-suite-build>`_.
+
+* Clone this repository to your local environment.
+
+* Run :bash:`make init` to install the dependencies.
+
+The example project
+-------------------
+
+The project contains:
+
+* ``.gitlab-ci.yml`` - Runs linting and tests in GitLab CI.
+* ``chipflow.toml`` - Congiuration telling the ChipFlow library how to load your Python design and allows you to configure the ChipFlow platform.
+* ``Makefile`` - Contains helpful shortcuts to the CLI tools used in the project.
+* ``my_design/chipflow.py`` - Used as a loader by the ChipFlow platform for the design.
+* ``my_design/design.py`` - This has the actual chip design.
+* ``my_design/my_contexts/*`` - These contexts control how your design will be presented to the ChipFlow contexts, ``simulation``(ulation), (FPGA)``board`` and ``silicon``.
+* ``my_design/sim/*`` - The C++ and `doit build file <https://pydoit.org/>`_ which builds the binary which will simulate our design.
+* ``my_design/software/*`` - The C++ and `doit build file <https://pydoit.org/>`_ for the software/BIOS which we will load onto our design when it's running in simulation or on a board.
+* ``tests/*`` - This has some pytest integration tests which cover the sim/board/silicon builds.
+
+The design
+----------
+
+The chip design is contained within the `MySoC` class in ``my_design/design.py``, and is described 
+using the `Amaranth hardware definition language <https://github.com/amaranth-lang/amaranth>`_.
+
+Something a bit unusual is that we change the implemention of some peripherals, 
+such as ``QSPIFlash``, according to the context the design is being used in. 
+
+Here's where we add the flash to our design:
+
+.. code-block:: python
+
+    self.rom = SPIMemIO(
+        flash=self.load_provider(platform, "QSPIFlash").add(m)
+    )
+
+The implementations, which are provided by ChipFlow, look a bit different for each context.
+
+QSPIFlash for a Board
+~~~~~~~~~~~~~~~~~~~~~
+
+For a board, in our case a ULX3S board, we need a means of accessing the clock pin (``USRMCLK``) and buffer primitives (``OBZ``, ``BB``) to join it together:
```

Comment:
suggestion "buffer primitives (``OBZ``, ``BB``) to access the other pins"

also adding "this is specific to the ECP5"  somewhere?